InTheHand.Net.Obex/ObexWebRequest.cs

```diff
@@ -664,6 +664,18 @@ private ObexStatusCode DoPut()
                 packetLength += 3 + headerValue.Length;
             }
 
+            // added David Rodgers
```

Comment:
As mentioned above I've added a more generic version of this to cycle through all the headers and add any UserN headers with their ObexHeader values (string only).

Reply:
As above, this is a better approach.